modules/search/search_results.go

```diff
@@ -0,0 +1,101 @@
+package search
+
+import (
+	"context"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"go.uber.org/atomic"
+)
+
+// SearchResults eases performing a highly parallel search by funneling all results into a single
+// channel that is easy to consume, signaling workers to quit early as needed, and collecting
+// metrics.
+type SearchResults struct {
+	resultsCh   chan *tempopb.TraceSearchMetadata
+	doneCh      chan struct{}
+	quit        atomic.Bool
+	workerCount atomic.Int32
+
+	tracesInspected atomic.Uint32
+	bytesInspected  atomic.Uint64
+}
+
+func NewSearchResults() *SearchResults {
```

Comment:
Unsure if this makes sense but we already have the worker pool code in `/tempodb/pool`. Perhaps it can be reused?

Reply:
pool.RunJobs waits for jobs to complete and returns the results synchronously.  That is not the direction I was going for, which is that the instance can end the search at any time because it has received enough results, or a deadline (timeout), and not wait for all tasks to complete.  For instance, a query of `name=xyz` could find enough results quickly and there is no reason to keep executing against the rest of the data.

Reply:
+1